pkgs/applications/editors/quartus-prime/default.nix

```diff
@@ -138,10 +138,43 @@ in buildFHSUserEnv {
     xorg.libXrender
   ];
 
-  extraInstallCommands = ''
-    mkdir -p $out/share/applications
-    cp ${desktopItem}/share/applications/* $out/share/applications
+  passthru = {
+    unwrapped = quartus;
+  };
+
+  extraInstallCommands = let
+    quartusExecutables = (map (c: "quartus/bin/quartus_${c}") [
+      "asm" "cdb" "cpf" "drc" "eda" "fit" "jbcc" "jli" "map" "pgm" "pow"
+      "sh" "si" "sim" "sta" "stp" "tan"
+    ]) ++ [ "quartus/bin/quartus" ];
+
+    qsysExecutables = map (c: "quartus/sopc_builder/bin/qsys-${c}") [
+      "generate" "edit" "script"
+    ];
+    # Should we install all executables ?
+    modelsimExecutables = map (c: "modelsim_ase/bin/${c}") [
+      "vsim" "vlog" "vlib"
+    ];
+  in ''
+    mkdir -p $out/share/applications $out/share/icons/128x128
+    ln -s ${desktopItem}/share/applications/* $out/share/applications
+    ln -s ${quartus}/licenses/images/dc_quartus_panel_logo.png $out/share/icons/128x128/quartus.png
+
+    mkdir -p $out/quartus/bin $out/quartus/sopc_builder/bin $out/modelsim_ase/bin
```

Comment:
Thank you @kwohlfahrt for accepting some of my changes. Personally, what I don't like most in the current way the executables are "installed" is that you create these directories. I can imagine myself a better way to integrate some the executables' names in a nix data type which will integrate nicely to the shell script, but I'm not experienced enough in nix to do it myself. Do you think you could write a shell script that would use something like this:

```nix
let
  executables = {
    quartus = {
      sharedPath = "quartus/bin";
      commands = (map (c: "quartus_${c}") [
        "asm" "cdb" "cpf" "drc" "eda" "fit" "jbcc" "jli" "map" "pgm" "pow"
        "sh" "si" "sim" "sta" "stp" "tan"
      ]) ++ [ "quartus/bin/quartus" ];
    };
    qsys = {
      sharedPath = "quartus/sopc_builder/bin";
	  commands = map (c: "qsys-${c}") [
	    "generate" "edit" "script"
      ];
	};
	modelsim = {
      sharedPath = "modelsim_ase/bin";
      commands = map (c: "${c}") [
        "vsim" "vlog" "vlib"
      ];
    };
  };
in


```
Perhaps?

This will also remove the need to link the directory at the end as well.

Reply:
Sure, that could definitely use some clean-up - I'll give that a go tomorrow. It was a deliberate choice to have both them and the links in `$out/bin` though - here is my reasoning:

Normally, quartus binaries get installed as `$QUARTUS_INSTALL_DIR/quartus/bin/...` (ans similarly for `modelsim_ase` and `quartus/sopc_builder`). So, some users might expect them to be in these relative directories instead of all in `$out/bin`. The way it is done now means that users can still use the same scripts for a normal quartus install and a nix build (just pointing `$QUARTUS_INSTALL_DIR` to the nix result). If we only keep the ones in `$out/bin` they will have to change their scripts to either look for these binaries in different places.

On the other hand, if we don't link to them from `$out/bin`, users installing quartus with `home-manager` or `nix-env` will not find the binaries on their `$PATH` as expected.

Does that seem a compelling enough use-case to keep both?

Reply:
> Normally, quartus binaries get installed as $QUARTUS_INSTALL_DIR/quartus/bin/... (ans similarly for modelsim_ase and quartus/sopc_builder) ...
> ...
> Does that seem a compelling enough use-case to keep both?

I guess so. Just for the record, it's not possible to run these executables without the wrapper right?

Reply:
They still need to go through `runScript`, because that sets up the environment with `/lib` & friends mounted in the expected place if that's what you meant?

Reply:
Yes.